### PR TITLE
Several bugs - changes for RHEL7 and RHEL6

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

File: foreman.if [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+# or any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see http://www.gnu.org/licenses/.
+#
```
Comment (lzap) [inline foreman.if:17]: I don't want to delete the file, I plan some interface extraction in the upcoming post 1.6 refactoring and split.

**Comment by lzap** [comment]
Do not merge yet, working on the bugs actually.

**Comment by lzap** [comment]
I am done with this, please review.

File: foreman.te [hunk 2]
```diff
@@ -192,8 +192,10 @@ optional_policy(`
 optional_policy(`
     tunable_policy(`passenger_run_foreman', `
         allow passenger_t self:process getsession;
+        allow passenger_t self:capability2 block_suspend;
```
Comment (lzap) [inline foreman.te]: From your bugreport I understand that Passenger wont load without this. I haven't tested. If it loads fine, we can just `dontaudit`...

Comment (domcleal) [inline foreman.te]: It didn't load at all, IIRC.

File: foreman.if [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
 
-	corecmd_search_bin($1)
-	domtrans_pattern($1, httpd_foreman_script_exec_t, httpd_foreman_script_t)
-')
```
Comment (lzap) [inline foreman.if:8]: Info for Jason - this bit was unused, so I am cleaning this up.

File: foreman.te [hunk 4]
```diff
@@ -251,6 +254,11 @@ optional_policy(`
     domtrans_pattern(passenger_t, puppetmaster_exec_t, puppetmaster_t)
 ')
 
+# Read and write in /var/run/foreman on RHEL7
+optional_policy(`
+    files_rw_inherited_generic_pid_files(passenger_t)
+')
+
```
Comment (lzap) [inline foreman.te]: This does not compile on RHEL6 and I don't know why - dont merge yet:

```
/usr/bin/checkmodule:  loading policy configuration from tmp/foreman.tmp
foreman.te":258:ERROR 'syntax error' at token 'files_rw_inherited_generic_pid_files' on line 11191:
#line 258
    files_rw_inherited_generic_pid_files(passenger_t)
```

File: foreman.te (hunk 2 above)
Comment (lzap) [inline foreman.te]: I confirmed with folks on #selinux that they allow this for some other base domains too, we should be fine. But this was not confirmed by Dan :-)

Comment (lzap) [inline foreman.te]: Ok I removed the rule bellow, it was not necessary. But the block_suspend does not compile on RHEL6 and I need to find some way to workaround this. I simple do not want to maintain two branches of this. I will preprocess the files with sed if there is no other way.

Comment (domcleal) [inline foreman.te]: Yeah, I'd prefer using a preprocessor of some sort over per-OS branching.

**Comment by lzap** [comment]
@domcleal I found a way how to modularize this without preprocessor using m4. see the Rakefile for example, I'd need to slightly change our SPEC tho. what you think?

**Comment by domcleal** [comment]
@lzap seems reasonable to me

**Comment by lzap** [comment]
Great, I think you can proceed to review, all three bugs should be covered now.

The packaging request is: https://github.com/theforeman/foreman-packaging/pull/308

File: foreman.te [hunk 5]
```diff
@@ -302,7 +306,10 @@ corenet_port(elasticsearch_port_t)
 # This is temporary solution until https://bugzilla.redhat.com/show_bug.cgi?id=1105085
 # is fixed.
 init_exec_script_files(passenger_t)
-consoletype_exec(passenger_t)
+
+ifdef(`distro_rhel7', `
```
Comment (domcleal) [inline foreman.te]: ifndef?

Comment (lzap) [inline foreman.te]: > > +ifdef(`distro_rhel7',`
> 
> ifndef?

Right it's the other way around, fixed.

## 

Later,
 Lukas #lzap Zapletal

**Comment by domcleal** [comment]
I'm getting errors during installation of the policy.  I additionally applied https://github.com/theforeman/foreman-packaging/pull/305 on top of your scratch build - here's mine: http://koji.katello.org/koji/taskinfo?taskID=139376.  I confirmed from build.log that it's setting the distro correctly.

The policy is failing around consoletype_exec_t.  Maybe the policy built with it, but the host I'm installing it onto doesn't have it?  I'm rather confused.

<pre>
[root@foremanrhel ~]# rpm -ivh "http://koji.katello.org/koji/getfile?taskID=139376&name=foreman-selinux-1.7.0-0.develop.el7.noarch.rpm"
Retrieving http://koji.katello.org/koji/getfile?taskID=139376&name=foreman-selinux-1.7.0-0.develop.el7.noarch.rpm
Preparing...                          ################################# [100%]
Updating / installing...
   1:foreman-selinux-1.7.0-0.develop.e################################# [100%]
libsepol.print_missing_requirements: foreman's global requirements were not met: type/attribute consoletype_exec_t (No such file or directory).
libsemanage.semanage_link_sandbox: Link packages failed (No such file or directory).
ValueError: Could not commit semanage transaction
ValueError: Type elasticsearch_port_t is invalid, must be a port type
warning: %post(foreman-selinux-1.7.0-0.develop.el7.noarch) scriptlet failed, exit status 1
</pre>

(with selinux-policy-3.12.1-153.el7_0.10.noarch)

edit: buildroot contains selinux-policy-3.12.1-153.el7.noarch, though I installed that and still had the same issue

**Comment by lzap** [comment]
Fixed now in the packaging PR, I was sending the DISTRO only to "clean" command.

I've pushed one another fix here - for relabel script. Do you see any other paths that could changed for RHEL7?

File: foreman-selinux-relabel [hunk 6]
```diff
@@ -5,6 +5,7 @@
   /usr/share/katello \
   /var/lib/foreman \
   /var/run/foreman \
+  /run/foreman \
```
Comment (domcleal) [inline foreman-selinux-relabel:8]: foreman.fc needs updating, it only contains /var/run/foreman

Comment (domcleal) [inline foreman-selinux-relabel:8]: Ignore that, there's an equivalency rule in place for /run and /var/run.

File: foreman.te [hunk 7]
```diff
@@ -222,6 +224,7 @@ optional_policy(`
         # Fedora.
         allow httpd_t puppet_etc_t:dir getattr;
```
Comment (domcleal) [inline foreman.te]: I additionally hit this:

<pre>
type=AVC msg=audit(1407836832.348:680): avc:  denied  { search } for  pid=14485 comm="httpd" name="puppet" dev="vda1" ino=18887 scontext=system_u:system_r:httpd_t:s0 tcontext=system_u:object_r:puppet_etc_t:s0 tclass=dir
</pre>

**Comment by domcleal** [comment]
Works well, thanks.  The only issue I hit was the puppet_etc_t one noted above.   /usr/lib64/gems/ruby/passenger-*/agents is the only other EL7-specific directory to relabel I think.

**Comment by lzap** [comment]
Thanks, pushed two additional commits to fix those, review and I will squash.

**Comment by domcleal** [comment]
:+1: OK to squash, thanks.  I left a comment on the packaging PR about exit codes.

**Comment by lzap** [comment]
Rebased, pushed! Thanks

**Comment by domcleal** [comment]
Thanks @lzap, merged as 7d5f1ef..e842477 for Foreman 1.6.0.

**Pull request closed without merging**